lib/Sema/TypeChecker.h

```diff
@@ -1520,6 +1520,12 @@ bool isMemberOperator(FuncDecl *decl, Type type);
 /// Complain if @objc or dynamic is used without importing Foundation.
 void diagnoseAttrsRequiringFoundation(SourceFile &SF);
 
+/// Returns `true` iff differentiable programming is enabled.
+bool isDifferentiableProgrammingEnabled(SourceFile &SF);
+
+/// Returns `true` iff `AdditiveArithmetic` derived conformances are enabled.
+bool areAdditiveArithmeticDerivedConformancesEnabled(SourceFile &SF);
```

Comment:
Starting with "are" is surely more grammatically correct, but it only seems to make this API less discoverable. APIs that start with "are" are usually ones that predicate on a plural noun phrase, but the "derived conformances" functionality is almost always treated as a single entity. I'd suggest making this API start with `is`, and for that purpose, changing the subject in the phrase to a singular noun phrase, e.g. "conformance derivation".

Reply:
Done in https://github.com/apple/swift/pull/30765/commits/51284a89b38922b1322edb34f55c516bcc05af49. I considered the same!